<launch>

  <!-- PX4 args -->
  <arg name="mav_name" default="iris"/>
  <arg name="fcu_url" default="udp://:14540@127.0.0.1:14557"/>
  <arg name="gcs_url" default="" />
  <arg name="tgt_system" default="1" />
  <arg name="tgt_component" default="1" />
  <arg name="command_input" default="2" />
  <arg name="gazebo_simulation" default="true" />
  <arg name="visualization" default="true"/>
  <arg name="log_output" default="screen" />
  <arg name="fcu_protocol" default="v2.0" />
  <arg name="respawn_mavros" default="false" />

  <!-- RISEQ args -->
  <arg name="gate_x" default="3"/>
  <arg name="gate_y" default="0"/>
  <arg name="gate_z" default="1.67"/>
  <arg name="gate_roll" default = "0.0"/>
  <arg name="gate_pitch" default = "0.0"/>
  <arg name="gate_yaw" default = "-1.5707"/>


  <!-- PX4 SITL, MAVROS, GAZEBO -->
  <include file="$(find mavros)/launch/px4.launch">
    <arg name="fcu_url" value="$(arg fcu_url)"/>
  </include>

  <include file="$(find px4)/launch/posix_sitl.launch">
      <arg name="vehicle" value="$(arg mav_name)"/>
  </include>

  <!-- DRL GATE and any other objects -->
  <include file="$(find riseq_gazebo)/launch/spawn_sdf.launch">
    <arg name="robot_name" default="drl_gate"/>
    <arg name="x" value="$(arg gate_x)"/>
    <arg name="y" value="$(arg gate_y)"/>
    <arg name="z" value="$(arg gate_z)"/>
    <arg name="roll" value="$(arg gate_roll)"/>
    <arg name="pitch" value="$(arg gate_pitch)"/>
    <arg name="yaw" value="$(arg gate_yaw)"/>
    <arg name="sdf_robot_file" default="$(find riseq_gazebo)/models/drl_gate/drl_gate.sdf"/>
  </include>


  <!-- RISEQ NAVIGATION launch -->

  <rosparam command="load" file="$(find riseq_common)/config/drones/iris.yaml"/>

  <param name="/riseq/gate_x" value="$(arg gate_x)" type="double"/>
  <param name="/riseq/gate_y" value="$(arg gate_y)" type="double"/>
  <param name="/riseq/gate_z" value="$(arg gate_z)" type="double"/>
  <param name="/riseq/gate_yaw" value="$(arg gate_yaw)" type="double"/>
  <param name="riseq/gate_pose_nn_cfg" value="$(find riseq_perception)/scripts/ss6p_gate/cfg/yolo-pose.cfg"/>
  <param name="riseq/gate_pose_nn_ply" value="$(find riseq_perception)/scripts/ss6p_gate/cfg/gate.ply"/>
  <param name="riseq/gate_pose_nn_wgt" value="$(find riseq_perception)/scripts/ss6p_gate/weights/model.weights"/>
  <param name="riseq/gate_pose_nn_data" value="$(find riseq_perception)/scripts/ss6p_gate/cfg/gate3.data"/>

  <node pkg = "riseq_perception" name ="riseq_gate_pose_publisher" type="riseq_gate_pose_publisher.py"/>
  <node name="riseq_high_level_control_publisher" pkg="riseq_control" type="riseq_high_level_control_publisher.py" output="screen"/>
  <!--node name="riseq_low_level_control_publisher" pkg="riseq_control" type="riseq_low_level_control_publisher.py"/-->
  <node name="rqt_plot" pkg="rqt_plot" type="rqt_plot"/>
  <!--node name="rviz" pkg="rviz" type="rviz"/-->
  <!--<node pkg="riseq_trajectory" name="riseq_MST_publisher" type="MST_publisher.py">
    <remap from="riseq/trajectory/uav_reference_trajectory" to="riseq/uav_trajectory"/>
  </node>-->
  <node pkg="riseq_tests" name="riseq_simple_trajectory_publisher" type="riseq_simple_trajectory_publisher.py" output="screen">
    <remap from="riseq/tests/uav_simple_trajectory" to="riseq/uav_trajectory" />
  </node>



</launch>